opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/jaeger_format.py

```diff
@@ -0,0 +1,129 @@
+    """
+
+    TRACE_ID_KEY = "uber-trace-id"
+    BAGGAGE_PREFIX = "uberctx-"
+
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> Context:
+
+        if context is None:
+            context = get_current()
+        fields = _extract_first_element(
+            getter.get(carrier, self.TRACE_ID_KEY)
+        ).split(":")
+        if len(fields) == 4:
+            trace_id, span_id, _parent_id, flags = fields
+        else:
+            return trace.set_span_in_context(trace.INVALID_SPAN)
+        span = trace.DefaultSpan(
+            trace.SpanContext(
+                trace_id=int(trace_id, 16),
+                span_id=int(span_id, 16),
+                is_remote=True,
+                trace_flags=trace.TraceFlags(
+                    int(flags) & trace.TraceFlags.SAMPLED
+                ),
+            )
+        )
+        baggage_keys = [
+            key
+            for key in getter.keys(carrier)
+            if key.startswith(self.BAGGAGE_PREFIX)
+        ]
+        for key in baggage_keys:
+            value = _extract_first_element(getter.get(carrier, key))
+            context = baggage.set_baggage(
+                key.replace(self.BAGGAGE_PREFIX, ""),
+                urllib.parse.unquote_plus(value).strip(),
+                context=context,
+            )
+        return trace.set_span_in_context(span, context)
+
+    def inject(
+        self,
+        set_in_carrier: Setter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> None:
+        span = trace.get_current_span(context=context)
+        span_context = span.get_span_context()
+        if span_context == trace.INVALID_SPAN_CONTEXT:
+            return
+
+        span_parent_id = span.parent.span_id if span.parent else 0
+        trace_flags = span_context.trace_flags
+        # set debug flag to True if sampled flag is set
+        if trace_flags.sampled:
+            trace_flags |= 0x02
+
+        # set span identity
+        set_in_carrier(
+            carrier,
+            self.TRACE_ID_KEY,
+            format_uber_trace_id(
+                span_context.trace_id,
+                span_context.span_id,
+                span_parent_id,
+                trace_flags,
+            ),
+        )
+
+        # set span baggage, if any
+        baggage_entries = baggage.get_all(context=context)
+        if not baggage_entries:
+            return
+        for key, value in baggage_entries.items():
+            baggage_key = self.BAGGAGE_PREFIX + key
+            set_in_carrier(
+                carrier, baggage_key, urllib.parse.quote_plus(str(value))
+            )
+
+
+def format_uber_trace_id(trace_id, span_id, parent_span_id, flags):
+    return "{:032x}:{:016x}:{:016x}:{:02x}".format(
+        trace_id, span_id, parent_span_id, flags
+    )
+
+
+def _extract_first_element(
```

Comment:
seems like this could just share the same extract_first_element that's used in the other propagators. is there value in this code being duplicated?

Reply:
Do you mean import it from B3 propagator or move it to a common file(perhaps init.py?) and import it from there?

Reply:
common would be best. that code isn't b3 propagator specific.

Reply:
Done, the b3 file as well as class needs to be changed anyway, so I ll raise a separate issue for that